src/utils/HttpClient.php

```diff
@@ -9,35 +9,69 @@
 final class HttpClient
 {
     /**
+     * @param non-empty-array<non-empty-string> $urls
      * @param list<string> $headers
-     * @return array{int, string}
+     *
+     * @return non-empty-array<non-empty-string, array{int, string}>
      */
-    public function get(string $url, array $headers): array
+    public function getMulti(array $urls, array $headers): array
     {
-        $curl = curl_init($url);
-        if (!$curl) {
-            throw new RuntimeException('Could not initialize cURL connection');
-        }
+        $mh = curl_multi_init();
 
-        // see https://stackoverflow.com/a/27776164/1597388
-        curl_setopt($curl, CURLOPT_CONNECTTIMEOUT, 3);
-        curl_setopt($curl, CURLOPT_TIMEOUT, 10);
+        $handles = [];
 
-        curl_setopt($curl, CURLOPT_FOLLOWLOCATION, true);
-        curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+        foreach ($urls as $url) {
+            $curl = curl_init($url);
+            if (!$curl) {
+                throw new RuntimeException('Could not initialize cURL connection');
+            }
 
-        if ([] !== $headers) {
-            curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
-        }
+            // see https://stackoverflow.com/a/27776164/1597388
+            curl_setopt($curl, CURLOPT_CONNECTTIMEOUT, 3);
+            curl_setopt($curl, CURLOPT_TIMEOUT, 10);
+
+            curl_setopt($curl, CURLOPT_FOLLOWLOCATION, true);
+            curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
 
-        $response = curl_exec($curl);
+            if ([] !== $headers) {
+                curl_setopt($curl, CURLOPT_HTTPHEADER, $headers);
+            }
+
+            $handles[$url] = $curl;
+        }
 
-        if (!is_string($response)) {
-            throw new RuntimeException("Could not fetch url $url");
+        foreach ($handles as $handle) {
+            curl_multi_add_handle($mh, $handle);
         }
 
-        $responseCode = curl_getinfo($curl, CURLINFO_RESPONSE_CODE);
+        do {
+            $status = curl_multi_exec($mh, $active);
+            if ($active) {
+                // Wait a short time for more activity
+                curl_multi_select($mh);
+            }
+        } while ($active && CURLM_OK == $status);
+
+        $result = [];
+        foreach ($handles as $url => $handle) {
+            $response = curl_multi_getcontent($handle);
+            $errno = curl_multi_errno($mh);
+
+            if ($errno || !is_string($response)) {
+                $message = curl_multi_strerror($errno);
+                if (null === $message) {
+                    $message = "Could not fetch url $url";
+                }
+                throw new RuntimeException($message);
+            }
+
+            $responseCode = curl_getinfo($handle, CURLINFO_RESPONSE_CODE);
+            $result[$url] = [$responseCode, $response];
+
+            curl_multi_remove_handle($mh, $handle);
```

Comment:
the curl PHP 7.4 phpstan errors are fixed with https://github.com/phpstan/phpstan-src/pull/2882 in the next PHPStan release